ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,462 @@
+            style: partial_u8_bits_to_u8(&raw_mouth_details[0..=5]),
+            color: partial_u8_bits_to_u8(&raw_mouth_details[6..=8]),
+            scale: partial_u8_bits_to_u8(&raw_mouth_details[9..=12]),
+            y_scale: partial_u8_bits_to_u8(&raw_mouth_details[13..=15]),
+        };
+
+        let mustache_details = MustacheDetails {
+            mouth_y_position: partial_u8_bits_to_u8(&raw_mustache_details[0..=4]),
+            mustache_style: partial_u8_bits_to_u8(&raw_mustache_details[5..=7]),
+        };
+
+        let beard_details = BeardDetails {
+            style: partial_u8_bits_to_u8(&raw_beard_details[0..=2]),
+            color: partial_u8_bits_to_u8(&raw_beard_details[3..=5]),
+            scale: partial_u8_bits_to_u8(&raw_beard_details[6..=9]),
+            y_position: partial_u8_bits_to_u8(&raw_beard_details[10..=14]),
+        };
+
+        let glass_details = GlassDetails {
+            style: partial_u8_bits_to_u8(&raw_glass_details[0..=3]),
+            color: partial_u8_bits_to_u8(&raw_glass_details[4..=6]),
+            scale: partial_u8_bits_to_u8(&raw_glass_details[7..=10]),
+            y_position: partial_u8_bits_to_u8(&raw_glass_details[11..=15]),
+        };
+
+        let mole_details = MoleDetails {
+            is_enabled: raw_mole_details[0],
+            scale: partial_u8_bits_to_u8(&raw_mole_details[1..=4]),
+            x_position: partial_u8_bits_to_u8(&raw_mole_details[5..=9]),
+            y_position: partial_u8_bits_to_u8(&raw_mole_details[10..=14]),
+        };
+
+        MiiData {
+            options,
+            selector_position: position,
+            console_identity: device,
+            system_id,
+            mac_address: creator_mac,
+            details,
+            name: mii_name,
+            height,
+            width,
+            face_details,
+            hair_details,
+            eye_details,
+            eyebrow_details,
+            nose_details,
+            mouth_details,
+            mustache_details,
+            beard_details,
+            glass_details,
+            mole_details,
+            author_name,
+        }
+    }
+}
+
+// Methods to handle "_bits_", ``bitvec`` cannot compile to 32-bit targets, so I had to create a few
+// helper methods
+
+/// Transforms a u8 into a [bool; 8]
+fn vec_bit(data: u8) -> [bool; 8] {
+    (0..8)
+        .map(|i| (data & (1 << i)) != 0)
+        .collect::<Vec<bool>>()
+        .try_into()
+        .unwrap()
+}
+
+/// Transforms a [bool; 8] into an u8
+fn vec_bit_to_u8(data: [bool; 8]) -> u8 {
+    data.into_iter()
+        .fold(0, |result, bit| (result << 1) ^ u8::from(bit))
+}
+
+/// Given a series of LE bits, they are filled until a full LE u8 is reached
+fn partial_u8_bits_to_u8(data: &[bool]) -> u8 {
+    let leading_zeroes_to_add = 8 - data.len();
+    let leading_zeroes = vec![false; leading_zeroes_to_add];
+
+    vec_bit_to_u8([data, &leading_zeroes].concat().try_into().unwrap())
```

Comment:
If these are leading zeros, why aren't they at the start of the list?

Reply:
Because of the way the bits are ordered by the 3ds, (this time I tested thoroughly to make sure the values returned are correct, so this should always work :p, but I'll double check later)

Reply:
Yeah. Tested again and it works properly